### ISSUE TITLE
Add support for pulling in additional dependencies through `externalReferences`

### ISSUE THREAD

**Issue by vpetersson** [DependencyTrack/dependency-track#4277]
### Current Behavior

It is my understanding that Dependency Track does not currently pull in dependencies specified in [externalReferences](https://cyclonedx.org/docs/1.6/json/#externalReferences). There are a lot of use cases for this, including more complex products.

### Proposed Behavior

The main use cases are `externalReferences` with URLs, in which case it would be great if DT could pull these in. Authentication is of course asn issue here.

`BOM-Link` is also an option in externalReferences, where it perhaps would make sense to have some kind of bundle (zip file?) with a set of SBOMs that can reference each other. 

### Checklist

- [x] I have read and understand the [contributing guidelines](https://github.com/DependencyTrack/dependency-track/blob/master/CONTRIBUTING.md#filing-issues)
- [x] I have checked the [existing issues](https://github.com/DependencyTrack/dependency-track/issues) for whether this enhancement was already requested

**Comment by valentijnscholten** [comment]
What do you mean by "pulling in"? I believe externalReferences are mainly used / meant to document systems, sites, and information that may be relevant but are not included with the BOM. Do you mean DT should pull in the contents of those urls to provide extra info? Or do you mean that when the external reference is of type BOM, it should be treated as a BOM and components should be ingested by DT?

**Comment by vpetersson** [comment]
What I mean is that if you have a "product SBOM", this would span multiple files. Thus externalRefernces is the only way to create a hierarchical SBOM structure, this is the most suitable tool for the job.

While I suppose you could just upload the SBOMs individually, having a DT to pull down all references would be desirable. This is also something that has come up in the discussions in [TEA](https://github.com/CycloneDX/transparency-exchange-api)/Project Koala.

**Comment by stevespringett** [comment]
Right. From an implementation perspective, DT would need to resolve any external reference of type `bom` and potentially pull in that inventory and dependencies as well. Ideally, this would automatically create a child project with those inventory and dependencies, but having an option to have in in the same project may also be useful in some cases.
